docs/USING_PUBLIC_CLOUDS.rst

```diff
@@ -0,0 +1,459 @@
+
+Edit this file to add a new region, which should describe the AWS S3 hosted
+bucket you'll be writing your data to. There are a few configurable options here:
+
+- :code:`type` : set to :code:`aws_s3` to indicate this region is hosted in AWS S3;
+- :code:`legacyAwsBehavior` : set to :code:`true` to indicate this region should
+  behave like AWS S3 :code:`us-east-1` region, set to :code:`false` to indicate
+  this region should behave like any other AWS S3 region;
+- :code:`bucketName` : set to an *existing bucket* in your AWS S3 Account; this
+  is the bucket that will host all the data written in this region;
+- :code:`awsEndpoint` : set to your bucket's endpoint, usually :code:`s3.amazonaws.com`;
+- :code:`bucketMatch` : set to :code:`true` if you want your object name to be the
+  same in your local bucket and your AWS S3 bucket; set to :code:`false` if you
+  want your object name to be of the form :code:`{{localBucketName}}/{{objectname}}`
+  in your AWS S3 hosted bucket;
+- :code:`credentialsProfile` and :code:`credentials` are two ways to provide
+  your AWS S3 credentials for that bucket, *use only one of them* :
+  - :code:`credentialsProfile` : set to the profile name allowing you to access
+    your AWS S3 bucket from your :code:`~/.aws/credentials` file;
+  - :code:`credentials` : set the two fields inside the object (:code:`accessKey`
+    and :code:`secretKey`) to their respective values from your AWS credentials.
+
+.. code:: json
+
+    (...)
+    "aws-test": {
+        "type": "aws_s3",
+        "legacyAwsBehavior": true,
+        "details": {
+            "awsEndpoint": "s3.amazonaws.com",
+            "bucketName": "zenkobucket",
+            "bucketMatch": true,
+            "credentialsProfile": "zenko"
+        }
+    },
+    (...)
+
+.. code:: json
+
+    (...)
+    "aws-test": {
+        "type": "aws_s3",
+        "legacyAwsBehavior": true,
+        "details": {
+            "awsEndpoint": "s3.amazonaws.com",
+            "bucketName": "zenkobucket",
+            "bucketMatch": true,
+            "credentials": {
+                "accessKey": "WHDBFKILOSDDVF78NPMQ",
+                "secretKey": "87hdfGCvDS+YYzefKLnjjZEYstOIuIjs/2X72eET"
+            }
+        }
+    },
+    (...)
+
+.. WARNING::
+   If you set :code:`bucketMatch` to :code:`true`, we strongly advise for having
+   only one local bucket per AWS S3 location. Indeed, since your objects names
+   won't be prefixed by the local bucket name in the AWS S3 bucket, you could
+   create inconsistencies seamlessly by putting two objects with the same name
+   in two different local buckets, but since they are ultimately saved in the
+   same AWS bucket, the most recent one would overwrite the earlier one, as the
+   namespace will conflict.
+
+config.json
+^^^^^^^^^^^
+
+Edit the :code:`restEndpoint` section of your :code:`config.json` file to add
+an endpoint definition matching your new AWS S3 hosted region. Following our
+previous example, it would look like:
```

Comment:
Hasn't this become obsolete for public clouds? I did a docker run without mounting config.json and could write to a custom AWS location... @LaurenSpiegel ?

Reply:
You have to modify your config.json if you want to add the new locationConstraint 'aws-test' as the default for a certain endpoint. I think it is confusing to use the endpoint 's3.amazonaws.com' and this addition is unnecessary.  As long as you have some default location for each of your endpoints, you don't need to add more just because you add a new location.

Reply:
OK, I added an "IMPORTANT" section specifying that you shouold only change config.json to change the default location to an endpoint. In this way, we document how to do it, and it should be less confusing. Let me know.